hazelcast-client/src/main/java/com/hazelcast/client/spi/ProxyManager.java

```diff
@@ -187,6 +190,29 @@ public ClientProxy create(String id) {
                 throw ExceptionUtil.rethrow(e);
             }
         }
+
+        readProxyDescriptors();
+    }
+
+    private void readProxyDescriptors() {
+        try {
+            ClassLoader classLoader = client.getClientConfig().getClassLoader();
+            Iterator<Class<ClientProxyDescriptorProvider>> iter =
+                    com.hazelcast.util.ServiceLoader.classIterator(PROVIDER_ID, classLoader);
+
+            while (iter.hasNext()) {
+                Class<ClientProxyDescriptorProvider> clazz = iter.next();
+                Constructor<ClientProxyDescriptorProvider> constructor = clazz.getDeclaredConstructor();
```

Comment:
It would be best to do this step by step and help out the user if something goes wrong. For example if the user fails to provide a no arg-constructor, there should be a clear message.